backend/src/test/groovy/pt/ulisboa/tecnico/socialsoftware/tutor/tournament/service/CancelTournamentTest.groovy

```diff
@@ -0,0 +1,115 @@
+package pt.ulisboa.tecnico.socialsoftware.tutor.tournament.service
+
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest
+import org.springframework.boot.test.context.TestConfiguration
+import pt.ulisboa.tecnico.socialsoftware.tutor.BeanConfiguration
+import pt.ulisboa.tecnico.socialsoftware.tutor.exceptions.TutorException
+import pt.ulisboa.tecnico.socialsoftware.tutor.question.domain.Assessment
+import pt.ulisboa.tecnico.socialsoftware.tutor.question.domain.Question
+import pt.ulisboa.tecnico.socialsoftware.tutor.tournament.dto.TournamentDto
+import pt.ulisboa.tecnico.socialsoftware.tutor.user.User
+
+import static pt.ulisboa.tecnico.socialsoftware.tutor.exceptions.ErrorMessage.*
+
+@DataJpaTest
+class CancelTournamentTest extends TournamentTest {
+    def tournamentDto
+
+    def setup() {
+        tournamentDto = new TournamentDto()
+        tournamentDto.setEndTime(STRING_DATE_LATER)
+        tournamentDto.setNumberOfQuestions(NUMBER_OF_QUESTIONS)
+        tournamentDto.setCanceled(false)
+
+        createAssessmentWithTopicConjunction(ASSESSMENT_1_TITLE, Assessment.Status.AVAILABLE, externalCourseExecution)
+
+        def question1 = createQuestion(LOCAL_DATE_TODAY, QUESTION_1_CONTENT, QUESTION_1_TITLE, Question.Status.AVAILABLE, externalCourse)
+
+        createOption(OPTION_1_CONTENT, question1)
+    }
+
+    def "user that created tournament cancels it"() {
+        given:
+        tournamentDto.setStartTime(STRING_DATE_TOMORROW)
+        tournamentDto = tournamentService.createTournament(user1.getId(), externalCourseExecution.getId(), topics, tournamentDto)
+
+        when:
+        tournamentService.cancelTournament(user1.getId(), tournamentDto)
+
+        then:
+        tournamentRepository.count() == 1L
+        def result = tournamentRepository.findAll().get(0)
+        result.isCanceled()
+    }
+
+    def "user that created an open tournament tries to cancel it"() {
+        given:
+        tournamentDto.setStartTime(STRING_DATE_TODAY)
+        tournamentDto = tournamentService.createTournament(user1.getId(), externalCourseExecution.getId(), topics, tournamentDto)
```

Comment:
idem, define a method called createTournament that is reused in all tests

Reply:
Done